src/redisai.c

```diff
@@ -426,27 +418,67 @@ int RedisAI_Set_Graph_RedisCommand(RedisModuleCtx *ctx, RedisModuleString **argv
     return RedisModule_ReplyWithError(ctx, "ERR unsupported device");
   }
 
-  RAI_Graph *graph = NULL;
+  long long argidx = 4;
+
+  char **inputs = array_new(char*, argc - argidx);
+  char **outputs = array_new(char*, argc - argidx);
+
+  size_t namelen;
+  const char *name, *option;
+
+  option = RedisModule_StringPtrLen(argv[argidx], NULL);
+
+  if (strcasecmp(option, "INPUTS") == 0) {
+    argidx++;
+    for (size_t i=0; i<argc-argidx-1; i++) {
+      name = RedisModule_StringPtrLen(argv[argidx + i], &namelen);
```

Comment:
It's safe to assume here that the return value (RedisModule_StringPtrLen) is NUL-terminated; no need for the extra alloc/copy here... and the function you're passing this into does not modify the strings.